tests/test_injection.py

```diff
@@ -250,6 +256,71 @@ def callback(response: DummyResponse,
         }
 
 
+class Html(Injectable):
+    url = "http://example.com"
+    html = """<html><body>Price: <span class="price">22</span>€</body></html>"""
+
+
+class EurDollarRate(Injectable):
+    rate = 1.1
+
+
+class OtherEurDollarRate(Injectable):
+    rate = 2
+
+
+@attr.s(auto_attribs=True)
+class PricePO(ItemPage, ResponseShortcutsMixin):
+    response: Html
+
+    def to_item(self):
+        return dict(price=float(self.css(".price::text").get()), currency="€")
+
+
+@attr.s(auto_attribs=True)
+class PriceInDollarsPO(ItemPage):
+    original_po: PricePO
+    conversion: EurDollarRate
+
+    def to_item(self):
+        item = self.original_po.to_item()
+        item["price"] *= self.conversion.rate
+        item["currency"] = "$"
+        return item
+
+
+class TestInjectorOverrides:
+
+    @pytest.mark.parametrize("override_expected", [True, False])
+    @inlineCallbacks
+    def test_overrides(self, providers, override_expected):
+        overrides = {("other-example.com", "example.com")[override_expected]: {
```

Comment:
I think it is better to extract the common code in other function and have a clear setup and expectations, instead of having to build the `if` paths based on the input

Reply:
I've tried to make it more clear here https://github.com/scrapinghub/scrapy-poet/pull/43/commits/222e950447159c2463a5aedae542519b2c419e7c